src/Filesystem/Definitions.php

```diff
@@ -12,22 +12,27 @@
 class Definitions
 {
 
+    /**
+     * Lock down the constructor for this class.
+     */
+    final public function __construct()
+    {
+    }
```

Comment:
Because `Definitions` is called statically, this ensures that static calls are safe even with extensions to the class.

Reply:
Can you give me an example of unsafe behaviour that this protects against? Apologies if you already have, probably just forgot.

Reply:
It probably will never be a problem for this class in particular, but since this change has been done to a few classes, I'll try and explain the situation better.

So with classes where we have static calls that create a new instance of the same class, like the `get` method in this class:

```php
    public static function get(string $type): array
    {
        return (new static)->getDefinitions($type);
    }
```

This method will break if someone were to extend the class and add an additional parameter to the constructor, because, as you see above, the method is returning a new instance of the class *without* any parameters. Basically, our static calls are expecting a certain signature for the constructor, which an extending class might break.

Making a constructor final prevents this, by enforcing a certain signature for the constructor, ensuring that the `new static` calls will always work.